wizard/class-instant-articles-wizard.php

```diff
@@ -0,0 +1,291 @@
+require_once( dirname( __FILE__ ) . '/class-instant-articles-wizard-state.php' );
+require_once( dirname( __FILE__ ) . '/class-instant-articles-wizard-fb-helper.php' );
+require_once( dirname( __FILE__ ) . '/class-instant-articles-wizard-review-submission.php' );
+
+use Facebook\InstantArticles\Client\Client;
+use Facebook\InstantArticles\Client\ClientException;
+
+/**
+* Controller for Set-up Wizard
+*
+* @since 3.1
+*/
+class Instant_Articles_Wizard {
+
+	public static function init() {
+		add_action( 'admin_menu', array( 'Instant_Articles_Wizard', 'menu_items' ) );
+
+		add_filter( 'plugin_action_links_' . IA_PLUGIN_PATH, array( 'Instant_Articles_Wizard', 'add_settings_link_to_plugin_actions' ) );
+
+		add_action( 'admin_init', function () {
+			new Instant_Articles_Option_FB_App();
+			new Instant_Articles_Option_FB_Page();
+			new Instant_Articles_Option_Styles();
+			new Instant_Articles_Option_Ads();
+			new Instant_Articles_Option_Analytics();
+			new Instant_Articles_Option_Publishing();
+		});
+
+		add_action(
+			'wp_ajax_instant_articles_wizard_transition',
+			array( 'Instant_Articles_Wizard', 'transition' )
+		);
+
+		add_action(
+			'wp_ajax_instant_articles_wizard_save_app',
+			array( 'Instant_Articles_Wizard', 'save_app' )
+		);
+
+		add_action(
+			'wp_ajax_instant_articles_wizard_edit_app',
+			array( 'Instant_Articles_Wizard', 'edit_app' )
+		);
+
+		add_action(
+			'wp_ajax_instant_articles_wizard_is_page_signed_up',
+			array( 'Instant_Articles_Wizard', 'is_page_signed_up' )
+		);
+
+		add_action(
+			'wp_ajax_instant_articles_wizard_submit_for_review',
+			array( 'Instant_Articles_Wizard', 'submit_for_review' )
+		);
+	}
+
+	public static function add_settings_link_to_plugin_actions( $links ) {
+		$link_text = __( 'Settings' );
+		$settings_href = self::get_url();
+		$settings_link = '<a href="' . esc_url( $settings_href ) . '">' . $link_text . '</a>';
+		array_push( $links, $settings_link );
+		return $links;
+	}
+
+	public static function menu_items() {
+		add_menu_page(
+			'Instant Articles Setup Wizard',
+			'Instant Articles',
+			'manage_options',
+			'instant-articles-wizard',
+			array( 'Instant_Articles_Wizard', 'render' )
+			,'dashicons-facebook'
+		);
+		// Hack to let the URL visible to ajax handlers
+		update_option( 'instant-articles-wizard-url', menu_page_url( 'instant-articles-wizard', false) );
+	}
+
+	public static function get_url() {
+		$url = menu_page_url( 'instant-articles-wizard', false );
+
+		// Needed when calling from ajax
+		if ( ! $url ) {
+			$url = get_option( 'instant-articles-wizard-url' );
+		}
+
+		return $url;
+	}
+
+
+	public static function get_admin_url() {
+		$url = parse_url( admin_url() );
+		return $url['host'];
+	}
+
+	public static function transition() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html( 'You do not have sufficient permissions to access this page.' ) );
+		}
+
+		$new_state = filter_input( INPUT_POST, 'new_state' );
```

Comment:
We should use WordPress' own sanitisation functions here with `$_POST['new_state']` rather than `filter_input`.